src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/DescriptorMetadata.java

```diff
@@ -26,24 +26,31 @@
 import java.util.List;
 
 import com.synopsys.integration.alert.common.descriptor.config.field.ConfigField;
+import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
+import com.synopsys.integration.alert.common.enumeration.DescriptorType;
 import com.synopsys.integration.util.Stringable;
 
-public class UIComponent extends Stringable {
+public class DescriptorMetadata extends Stringable {
     private final String label;
     private final String urlName;
-    private final String descriptorName;
+    private final String name;
+    private final DescriptorType type;
+    private final ConfigContextEnum context;
     private final String fontAwesomeIcon;
     private final boolean automaticallyGenerateUI;
     private List<ConfigField> fields;
 
-    public UIComponent(final String label, final String urlName, final String descriptorName, final String fontAwesomeIcon, final List<ConfigField> fields) {
-        this(label, urlName, descriptorName, fontAwesomeIcon, true, fields);
+    public DescriptorMetadata(final String label, final String urlName, final String name, final DescriptorType type, final ConfigContextEnum context, final String fontAwesomeIcon, final List<ConfigField> fields) {
```

Comment:
DescriptorMetadata seems too generic, would you be against DescriptorFieldMetadata?